### ISSUE TITLE
Octave exampleexperiment.m crashes under Linux

### ISSUE THREAD

**Issue by brockho** [numbbo/coco#478]
In order to prepare the continuous integration test for Octave, I run `python do.py run-octave` on our linux slave. The experiment started nicely, but then crashed because of a wrong assertion wrt. the Schwefel function:
```
Started by upstream project "Coco-experiments-linux-dimo" build number 14
originally caused by:
 Started by user dimos@email.fr
Building remotely on numbbo-ubuntu-12p04-i386 in workspace /builds/disk/workspace/Coco-experiments-linux-dimo/dotype/run-octave/label/numbbo-ubuntu-12p04-i386
Cloning the remote Git repository
Cloning repository https://github.com/brockho/numbbo/
 > git init /builds/disk/workspace/Coco-experiments-linux-dimo/dotype/run-octave/label/numbbo-ubuntu-12p04-i386 # timeout=10
Fetching upstream changes from https://github.com/brockho/numbbo/
 > git --version # timeout=10
using GIT_SSH to set credentials 
using .gitcredentials to set credentials
 > git config --local credential.username NumbboCI # timeout=10
 > git config --local credential.helper store --file=/tmp/git5264399344173621038.credentials # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/brockho/numbbo/ +refs/heads/*:refs/remotes/origin/*
 > git config --local --remove-section credential # timeout=10
 > git config remote.origin.url https://github.com/brockho/numbbo/ # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/brockho/numbbo/ # timeout=10
Fetching upstream changes from https://github.com/brockho/numbbo/
using GIT_SSH to set credentials 
using .gitcredentials to set credentials
 > git config --local credential.username NumbboCI # timeout=10
 > git config --local credential.helper store --file=/tmp/git1342965195315106853.credentials # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/brockho/numbbo/ +refs/heads/*:refs/remotes/origin/*
 > git config --local --remove-section credential # timeout=10
Checking out Revision 050b50976e15e8dc3074e0fe015fb5f036f3eecf (refs/remotes/origin/dimo-develop)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 050b50976e15e8dc3074e0fe015fb5f036f3eecf
First time build. Skipping changelog.
[numbbo-ubuntu-12p04-i386] $ /bin/sh -xe /tmp/hudson6512097224949721395.sh
+ python do.py run-octave
CLEAN	 mex files from code-experiments/build/matlab/
AML	['code-experiments/src/coco_generics.c', 'code-experiments/src/coco_random.c', 'code-experiments/src/coco_suite.c', 'code-experiments/src/coco_observer.c', 'code-experiments/src/coco_runtime_c.c'] -> code-experiments/build/matlab/coco.c
COPY	code-experiments/src/coco.h -> code-experiments/build/matlab/coco.h
WRITE	code-experiments/build/matlab/REVISION
WRITE	code-experiments/build/matlab/VERSION
RUN	octave --no-gui setup.m in code-experiments/build/matlab
RUN	octave --no-gui exampleexperiment.m in code-experiments/build/matlab
ERROR: return value=-6
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
warning: docstring file '/usr/local/share/octave/4.0.0/etc/built-in-docstrings' not found
COCO INFO: Results will be output to folder RS_on_bbob-biobj
Running random search on the bbob-biobj suite...

COCO INFO: 22.01.16 22:26:05, d=2, running: f01..........f02..........f03..........f04..........
f05..........f06..........f07..........f08..........f09..........f10..........f11..........
f12..........f13..........f14..........f15..........f16..........f17..........f18..........
f19..........f20..........f21..........f22..........f23..........f24..........f25..........
f26..........f27..........f28..........f29..........f30..........f31..........f32..........
f33..........f34..........f35..........f36..........f37..........f38..........f39..........
f40..........f41..........f42..........f43..........f44..........f45..........f46..........
f47..........f48..........f49..........f50..........f51..........f52..........f53..........
f54..........f55..........done
COCO INFO: 22.01.16 22:26:08, d=3, running: f01..........f02..........f03..........f04..........f05..........f06..........f07.........
f08..........octave-cli-4.0.0: code-experiments/src/f_schwefel.c:43:
f_schwefel_evaluate: Assertion `y[0] >= self->best_value[0]' failed.
panic: Aborted -- stopping myself...
attempting to save variables to 'octave-workspace'...
save to 'octave-workspace' complete

Traceback (most recent call last):
  File "do.py", line 647, in <module>
    main(sys.argv[1:])
  File "do.py", line 630, in main
    elif cmd == 'run-octave': run_octave()    
  File "do.py", line 411, in run_octave
    run('code-experiments/build/matlab', ['octave', '--no-gui', 'exampleexperiment.m'])
  File "/builds/disk/workspace/Coco-experiments-linux-dimo/dotype/run-octave/label/numbbo-ubuntu-12p04-i386/code-experiments/tools/cocoutils.py", line 103, in run
    universal_newlines=True)
  File "/usr/lib/python2.7/subprocess.py", line 544, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['octave', '--no-gui', 'exampleexperiment.m']' returned non-zero exit status -6
Build step 'Execute shell' marked build as failure
Sending e-mails to: dimos@email.fr
Finished: FAILURE
```

**Comment by ttusar** [comment]
Is this the same assertion that failed also sometimes before?
I really need to throw a `coco_error` instead of making an assertion at this point so that we can see what is happening. It could just be an error stemming from comparing doubles...

**Comment by ttusar** [comment]
So, I did this (replaced all `y[0] >= self->best_value[0]` assertions with throwing a `coco_error`), and immediately (and for the first time) an integration test failed:

````
COCO FATAL ERROR: f_schwefel_evaluate(): Found value (1.705302565824240e-15) better than best (1.705302565824240e-15)
````
This happened in the test that checks whether all `bbob` problems can be found. So it's a precision thing...

**Comment by nikohansen** [comment]
The problem is very probably that the value in the code is not correct. I will fix it.

**Comment by ttusar** [comment]
?

**Comment by ttusar** [comment]
Fixed by commit ed20fb7. Closing the issue.

**Comment by nikohansen** [comment]
See also #485.
